txnkv/transaction/txn.go

```diff
@@ -656,14 +663,19 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 			}
 			return err
 		}
+
+		if lockCtx.CheckExistence {
+			checkedExistence = true
+		}
 	}
 	for _, key := range keys {
 		valExists := tikv.SetKeyLockedValueExists
 		// PointGet and BatchPointGet will return value in pessimistic lock response, the value may not exist.
 		// For other lock modes, the locked key values always exist.
-		if lockCtx.ReturnValues {
+		if lockCtx.ReturnValues || checkedExistence {
 			val := lockCtx.Values[string(key)]
-			if len(val.Value) == 0 {
+			// TODO: Check if it's safe to use `val.Exists` instead of assuming empty value.
```

Comment:
I think for TiDB they are equivalent. Maybe `len(val.Value) == 0 || !val.Exists` in favor of safety?

Reply:
It doesn't seem to work if ReturnValue is false but CheckExistence is true?